src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs

```diff
@@ -277,9 +278,9 @@ Task SetAsync(string name, object value, Type declaredType)
         }
         else
         {
-            RunSubmitCodeLocally(code);
+            var success = RunSubmitCodeLocally(code);
 
-            if (pwsh.HadErrors)
+            if (!success || pwsh.HadErrors)
```

Comment:
Consider returning `false` from `RunSubmitCodeLocally` when `pwsh.HadErrors` is `true` so that we can just check for `!success` over here.